syntaxes/solidity.json

```diff
@@ -0,0 +1,540 @@
+            "captures": {
+                "1": {
+                    "name": "storage.type.constructor.solidity"
+                }
+            }
+        },
+        "declaration-enum": {
+            "match": "\\b(enum)(\\s+([A-Za-z_]\\w*))?\\b",
+            "captures": {
+                "1": {
+                    "name": "storage.type.enum.solidity"
+                },
+                "3": {
+                    "name": "entity.name.type.enum.solidity"
+                }
+            }
+        },
+        "declaration-function": {
+            "patterns": [
+                {
+                    "match": "\\b(function)\\s+([A-Za-z_]\\w*)\\b",
+                    "captures": {
+                        "1": {
+                            "name": "storage.type.function.solidity"
+                        },
+                        "2": {
+                            "name": "entity.name.function.solidity"
+                        }
+                    }
+                },
+                {
+                    "match": "\\b(private|public|internal|external|constant|pure|view|payable|nonpayable|inherited|indexed|storage|memory|virtual|override)\\b",
+                    "name": "storage.type.mofifier.solidity"
+                }
+            ]
+        },
+        "declaration-modifier": {
+            "match": "\\b(modifier)(\\s+([A-Za-z_]\\w*))?\\b",
+            "captures": {
+                "1": {
+                    "name": "storage.type.modifier.solidity"
+                },
+                "3": {
+                    "name": "entity.name.function.solidity"
+                }
+            }
+        },
+        "declaration-mapping": {
+            "match": "\\b(mapping)\\b",
+            "name": "storage.type.mapping.solidity"
+        },
+        "function-call": {
+            "match": "\\b([A-Za-z_]\\w*)\\s*\\(",
+            "captures": {
+                "1": {
+                    "name": "entity.name.function.solidity"
+                }
+            }
+        },
+        "assembly": {
+            "patterns": [
+                {
+                    "match": "\\b(assembly)\\b",
+                    "name": "keyword.control.assembly.solidity"
+                },
+                {
+                    "match": "\\b(let)\\b",
+                    "name": "storage.type.assembly.solidity"
+                }
+            ]
+        },
+        "punctuation": {
+            "patterns": [
+                {
+                    "match": ";",
+                    "name": "punctuation.terminator.statement.solidity"
+                },
+                {
+                    "match": "\\.",
+                    "name": "punctuation.accessor.solidity"
+                },
+                {
+                    "match": ",",
+                    "name": "punctuation.separator.solidity"
+                }
+            ]
+        }
+    },
+    "scopeName": "source.solidity",
+    "uuid": "123"
+}
```

Comment:
This file needs proper attribution. We also need to know what version was copied into our repo, so we can update it. Please refer to the exact commit of the source repo and include copyright and license information.